cl/corpus_importer/management/commands/invoice_project.py

```diff
@@ -49,9 +50,11 @@ def get_attachment_pages(options):
         {"rows": page_size, "fl": ["id", "docket_id"]},
         {"group": False, "facet": False, "highlight": False},
     )
-    si = ExtraSolrInterface(settings.SOLR_RECAP_URL, mode="r")
-    results = si.query().add_extra(**main_query)
-    si.conn.http_connection.close()
+    with Session() as session:
+        si = ExtraSolrInterface(
+            settings.SOLR_RECAP_URL, http_connection=session, mode="r"
+        )
+        results = si.query().add_extra(**main_query)
```

Comment:
This gets paginated later, which feels like trouble again?

Reply:
I figured it was best to do this refactoring while avoiding any logic changes regarding when sessions should close for the most part.

Reply:
OK re refactorings later. Some small ones in PRs like this seem fine to me, but I appreciate the diligence.